include/eosio/vm/types.hpp

```diff
@@ -189,11 +189,137 @@ namespace eosio { namespace vm {
       // to memory with static storage duration.
       const char *             error            = nullptr;
 
+      // Stores data needed by JIT execution, in memory managed by standard
+      // C++ vectors, not growable_allocator used by guarded_vector,
+      // such that growable_allocator can be released after parsing.
+      // This is to make it possible parsing WASM only once for JIT.
+      struct jit_mod_t {
+         struct jit_func_type {
+            value_type                 form;
+            std::vector<value_type>    param_types;
+            uint8_t                    return_count;
+            value_type                 return_type;
+         };
+         struct jit_import_type {
+            uint32_t    func_t;
+         };
+         struct jit_import_entry {
+            std::vector<uint8_t> module_str;
+            std::vector<uint8_t> field_str;
+            external_kind        kind;
+            jit_import_type      type;
+         };
+         struct jit_export_entry {
+            std::vector<uint8_t> field_str;
+            external_kind        kind;
+            uint32_t             index;
+         };
+         struct jit_data_segment {
+            uint32_t              index;
+            init_expr             offset;
+            std::vector<uint8_t>  data;
+         };
+
+         std::vector<jit_func_type>        types;
+         std::vector<jit_import_entry>     imports;
+         std::vector<uint32_t>             functions;
+         std::vector<memory_type>          memories;
+         std::vector<global_variable>      globals;
+         std::vector<jit_export_entry>     exports;
+         std::vector<size_t>               code_offset;
+         std::vector<jit_data_segment>     data;
+         std::vector<uint32_t>             import_functions;
```

Comment:
Why no `fast_functions`? I see why no `type_aliases`, but not as obvious why no `fast_functions`. Is it because they are only used by eos-vm and not eos-vm-jit?